test/templates/template_proxy_request_test.rb

```diff
@@ -0,0 +1,58 @@
+require 'test_helper'
+require 'uri'
+require 'net/http'
+require 'mocha'
+require 'templates/templates_plugin'
+require 'templates/template_proxy_request'
+require 'webmock/test_unit'
+
+
+class TemplateProxyRequestTest < Test::Unit::TestCase
+  def setup
+    @foreman_url = 'https://foreman.example.com'
+    Proxy::SETTINGS.stubs(:foreman_url).returns(@foreman_url)
+    @template_url = 'http://proxy.lan:8443'
+    Proxy::Templates::Plugin.settings.stubs(:template_url).returns(@template_url)
+    @request = mock('request')
```

Comment:
This mocking wouldn't be necessary if you passed environment and ip parameters directly into `TemplateProxyRequest.get_template` method. Not only would this make it clear what the method operates on, but would also make tests less fragile. Don't use mocks/stubs unless absolutely necessary.